src/ufuncs.c

```diff
@@ -1942,6 +1942,79 @@ static void polygonize_full_func(char** args, npy_intp* dimensions, npy_intp* st
 }
 static PyUFuncGenericFunction polygonize_full_funcs[1] = {&polygonize_full_func};
 
+static char nearest_points_dtypes[3] = {NPY_OBJECT, NPY_OBJECT, NPY_OBJECT};
+static void nearest_points_func(char** args, npy_intp* dimensions, npy_intp* steps,
+                                void* data) {
+  GEOSGeometry* in1 = NULL;
+  GEOSGeometry* in2 = NULL;
+  GEOSPreparedGeometry* in1_prepared = NULL;
+  GEOSGeometry** geom_arr;
+  GEOSCoordSequence* coord_seq = NULL;
+
+  CHECK_NO_INPLACE_OUTPUT(2);
+
+  // allocate a temporary array to store output GEOSGeometry objects
+  geom_arr = malloc(sizeof(void*) * dimensions[0]);
+  CHECK_ALLOC(geom_arr);
+
+  GEOS_INIT_THREADS;
+
+  BINARY_LOOP {
+    /* get the geometries: return on error */
+    if (!get_geom_with_prepared(*(GeometryObject**)ip1, &in1, &in1_prepared)) {
+      errstate = PGERR_NOT_A_GEOMETRY;
+      destroy_geom_arr(ctx, geom_arr, i - 1);
+      break;
+    }
+    if (!get_geom(*(GeometryObject**)ip2, &in2)) {
+      errstate = PGERR_NOT_A_GEOMETRY;
+      destroy_geom_arr(ctx, geom_arr, i - 1);
+      break;
+    }
+
+    if ((in1 == NULL) | (in2 == NULL)) {
+      // in case of a missing value: return NULL (None)
+      geom_arr[i] = NULL;
+    } else if (GEOSisEmpty_r(ctx, in1) | GEOSisEmpty_r(ctx, in2)) {
```

Comment:
This could be simplified to something like:
```
if ((in1 == NULL) | (in2 == NULL) | GEOSisEmpty_r(ctx, in1) | GEOSisEmpty_r(ctx, in2)) {
    // in case of a missing value or empty geometry: return NULL (None)
    // GEOSNearestPoints_r returns NULL for empty geometries
    // but this is not distinguishable from an actual error, so we handle this ourselves
    geom_arr[i] = NULL;
    continue;
}
#if GEOS_SINCE_3_9_0
```
(note: drops the corresponding `else` since the first conditional has a `continue`)